packages/liveblocks-node/src/client.ts

```diff
@@ -194,6 +293,377 @@ export class Liveblocks {
+    await this.post(url`/v2/rooms/${roomId}/broadcast_event`, message);
+  }
+
+  /* -------------------------------------------------------------------------------------------------
+   * Storage
+   * -----------------------------------------------------------------------------------------------*/
+
+  /**
+   * Returns the contents of the room’s Storage tree.
+   * The default outputted format is called “plain LSON”, which includes information on the Live data structures in the tree.
+   * These nodes show up in the output as objects with two properties:
+   *
+   * ```json
+   * {
+   *   "liveblocksType": "LiveObject",
+   *   "data": ...
+   * }
+   * ```
+   *
+   * If you’re not interested in this information, you can use the `format` parameter to get a more compact output.
+   *
+   * @param roomId The id of the room to get the storage from.
+   * @param format (optional) Set to return `plan-lson` representation by default. If set to `json`, the output will be formatted as a simplified JSON representation of the Storage tree.
+   * In that format, each LiveObject and LiveMap will be formatted as a simple JSON object, and each LiveList will be formatted as a simple JSON array. This is a lossy format because information about the original data structures is not retained, but it may be easier to work with.
+   */
+  public getStorageDocument(
+    roomId: string,
+    format: "plain-lson"
+  ): Promise<PlainLsonObject>;
+
+  public getStorageDocument(roomId: string): Promise<PlainLsonObject>; // Default to 'plain-lson' when no format is provided
+
+  public getStorageDocument(
+    roomId: string,
+    format: "json"
+  ): Promise<JsonObject>;
+
+  public async getStorageDocument(
+    roomId: string,
+    format: "plain-lson" | "json" = "plain-lson"
+  ): Promise<PlainLsonObject | JsonObject> {
+    const path = url`v2/rooms/${roomId}/storage`;
+    const res = await this.get(path, { format });
+    return (await res.json()) as Promise<PlainLsonObject | JsonObject>;
+  }
+
+  /**
+   * Initializes a room’s Storage. The room must already exist and have an empty Storage.
+   * Calling this endpoint will disconnect all users from the room if there are any.
+   *
+   * @param roomId The id of the room to initialize the storage from.
+   * @param document The document to initialize the storage with.
+   * @returns The initialized storage document. It is of the same format as the one passed in.
+   */
+  public async initializeStorageDocument(
+    roomId: string,
+    document: PlainLsonObject
+  ): Promise<PlainLsonObject> {
+    const res = await this.post(url`/v2/rooms/${roomId}/storage`, document);
+    return (await res.json()) as Promise<PlainLsonObject>;
+  }
+  /**
+   * Deletes all of the room’s Storage data and disconnect all users from the room if there are any.
+   * @param roomId The id of the room to delete the storage from.
+   */
+  public async deleteStorage(roomId: string): Promise<void> {
```

Comment:
Missed this the other day! Can we make this one `deleteStorageDocument`? Our docs currently have the wrong title—this should match the other two Storage functions.

Reply:
Also maybe mention explicitly that it will **not** delete the YDoc if it exists.

Reply:
Done! Updated the method name to `deleteStorageDocument` and updated the comment to say `Note that this does not delete the Yjs document in the room if one exists.`. Thanks @nvie @CTNicholas 

https://github.com/liveblocks/liveblocks/blob/ea2b50212f51c08eba046da122ed4c45b01438c4/packages/liveblocks-node/src/client.ts#L511C1-L517C4


```ts
/**
 * Deletes all of the room’s Storage data and disconnect all users from the room if there are any. Note that this does not delete the Yjs document in the room if one exists.
 * @param roomId The id of the room to delete the storage from.
 */
public async deleteStorageDocument(roomId: string): Promise<void> {
  await this.delete(url`/v2/rooms/${roomId}/storage`);
}
```